### PR TITLE
Fix force_soft_failure

### PR DESCRIPTION
**Pull request by asdil12**
Allow overwriting the test result

### CODE REVIEW

**Comment by asdil12** [comment]
Normal fail: http://artemis.suse.de/tests/1274
Forced softfail: http://artemis.suse.de/tests/1271#step/prepare_test_data/42

Seems to work fine.

**Comment by asdil12** [comment]
The only thing that I found is that the overall status will still be set to *failed*:
http://artemis.suse.de/tests/1287

@okurz WDYT? Is this expected? The documentation doesn't cover this one.

File: basetest.pm [hunk 1]
```diff
@@ -320,11 +320,12 @@ sub post_run_hook {
 sub run_post_fail {
     my ($self, $msg) = @_;
     return if $bmwqemu::vars{_SKIP_POST_FAIL_HOOKS};
+    $self->{result}                 = 'fail' if $self->{result};
```
Comment (SergioAtSUSE) [inline basetest.pm]: you added the result 'fail' here. Maybe you can use 'softfail'.

See https://github.com/os-autoinst/os-autoinst/blob/2bd444f4840a223602021e564d092fd6dc4ac69b/basetest.pm#L460

Comment (asdil12) [reply basetest.pm]: I don't want to change the existing behaviour.
I just want to allow overwriting the result from post fail hook.

**Comment by asdil12** [comment]
@coolo You can already merge - the overall status issue can be discussed and improved later - maybe with a parameter `force_overall_result`.

**Comment by okurz** [comment]
Sorry, I'm not really getting what you want to achieve here. I guess unit tests would help to explain this :)

**Comment by asdil12** [comment]
@okurz At the moment `force_soft_failure` will not set the result of a test module to `softfail` insted of `fail` as the result will be overwritten to `fail` a few lines later.
My patch will prevent this by executing that line earlier so that an overwrite via `force_soft_failure` will have an effect.

**Comment by okurz** [comment]
ok, then please add a test for that

**Comment by okurz** [comment]
@asdil12 can you please add tests?

**Pull request closed without merging**